Pobeg_lab_3/Pobeg_lab_3/Source.cpp

```diff
@@ -0,0 +1,258 @@
+#include<iostream>
+#include<windows.h>
+#include<vector>
+#include <cstdlib>
+#include<stdio.h>
+
+typedef struct parameter{
+	HANDLE* break_event;
+	HANDLE* stop_event;
+	int number;
+}parameter;
+
+CRITICAL_SECTION cs;
+CRITICAL_SECTION cs_console;
+HANDLE start_event;
+
+DWORD doMarkerWork(LPVOID);
+bool createThreads(std::vector<HANDLE>& threads, std::vector<parameter>& parameters,HANDLE* break_events, HANDLE* stop_events, int size,int number_of_threads);
+bool work_with_threads(HANDLE* stop_events, HANDLE* break_events, std::vector<HANDLE>& threads_vector, int number_of_threads, int size);
+void init_mass(HANDLE* mass_events,const char* temp_name, int number_of_threads);
+bool set_event(HANDLE* h_event, const char* name);
+void print(std::vector<int>&, int);
+void change_number(int index, int value);
+bool free_resources(HANDLE* stop_events, HANDLE* break_events);
+int get_thread_index(int number_of_threads, bool* flags_of_broken_threads);
+
+const int MY_NUMBER = 11;
+std::vector<int> numbers;
+
+int main()
+{	
+	int size;
+	std::cout << "enter size of massive:\n";
+	std::cin >> size;
+	numbers.resize(size);
+	
+	int number_of_threads;
+	std::cout << "enter number of threads:\n";
+	std::cin >> number_of_threads;
+	
+	std::vector<HANDLE>threads_vector;
+	threads_vector.reserve(number_of_threads);
+	HANDLE* break_events = new HANDLE[number_of_threads];
+	HANDLE* stop_events = new HANDLE[number_of_threads];
+
+	std::vector<parameter>parameters;
+	parameters.reserve(number_of_threads);
+	init_mass(break_events, "break_event_", number_of_threads);
+	init_mass(stop_events, "stop_event_", number_of_threads);
+
+	InitializeCriticalSection(&cs);
+	InitializeCriticalSection(&cs_console);
+
+	start_event = CreateEvent(nullptr, TRUE, FALSE, TEXT("start_event"));
+
+	threads_vector.reserve(number_of_threads);
+	if (!createThreads(threads_vector, parameters, break_events, stop_events,size,  number_of_threads))
+	{
+		free_resources(stop_events, break_events);
+		return -1;
+	}
+	
+	
+	if (!set_event(&start_event,"start_event"))
+	{
+		free_resources(stop_events, break_events);
+		return -1;
```

Comment:
use different return codes in different cases. Also, positive codes are usually used